source/zip.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+/**
+ * Объединяет свойства нескольких объектов в один объект
+ *
+ * @param {...object} objects - Объекты, свойства которых нужно объединить
+ * @throws {Error} Если хотя бы один из переданных параметров не является объектом (или же является массивом)
+ * @returns {object} Объединенный объект
+ */
+const zip = (...objects) => {
+    return objects.reduce((mergedObject, obj) => {
+        if (typeof obj !== 'object' || !obj || Array.isArray(obj)) {
+            throw new Error('The function only accepts objects');
```

Comment:
```suggestion
            throw new Error('The function only accepts array of objects');
```

Кажется, так нагляднее. А еще можно throw'ать [это](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError)